app/controllers/application_controller.rb

```diff
@@ -5,6 +5,14 @@ class ApplicationController < ActionController::Base
   before_action :set_locale
   before_action :configure_permitted_parameters, if: :devise_controller?
 
+  rescue_from CanCan::AccessDenied do |exception|
+    respond_to do |format|
+      format.json{head :forbidden, content_type: "text/html"}
+      format.html{redirect_to main_app.root_url, alert: exception.message}
```

Comment:
main_app.root_url
chỗ xử lý này mình dùng root_path có được ko ? vì sao ?

Reply:
exception.message
phần này nếu đưa ra view của end user thì có trực quan dễ hiểu cho họ ko ? nếu có i18n thì sao

Reply:
1. main_app.root_url có thể dùng path được nhé. 
   + _path: cung cấp đường dẫn tương đối
   + _url: cung cấp đường dẫn tuyệt đối (thường sử dụng cho trường hợp gửi link sang domain khác)
=> cả 2 cái đều redirect đến root do mình định nghĩa trong routes.
2. exception.message : phần này để "Exception Handling"
  + Mặc định nếu ko xử lý thì khi user ko có quyền truy cập vào 1 trang nào đó thì sẽ raiser lỗi : CanCan::AccessDenied in     ...(Bad User eXperience)
  + Để xử lý vấn đề này thì ta làm như trên để hiển thị message thông báo người dùng hoặc điều hướng đến 1 trang thông báo lỗi. Anh quên i18n để fix lại nhé.